app/modules/code_provider/github/github_service.py

```
@@ -873,6 +858,174 @@ def _format_node(node: Dict[str, Any], depth: int = 0) -> List[str]:
         return "\n".join(_format_node(structure))
 
+    async def get_repo_structure_json(
+        self, repo_name: str, branch_name: str
+    ) -> List[Dict[str, Any]]:
+        """
+        Get repository structure as JSON for a specific branch.
+        Uses Git Tree API for efficient single-call fetching of entire repo structure.
+        Falls back to recursive fetching if Git Tree API fails.
+        """
+        try:
+            github, repo = self.get_repo(repo_name)
```

Ruff (0.14.8)

Unpacked variable `github` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)